### PR TITLE
Use client certificate instead of client secret for the certificate rotation job

### PR DESCRIPTION
**Pull request by SvetlinZarev-SAP**
**Description**

use client certificate instead of client secret for the certificate rotation job

**Pull Request status**

- [X] Implementation
- [ ] Unit tests
- [ ] Integration tests
- [ ] `chart/compass/values.yaml` is updated 
- [ ] Mocks are regenerated, using the automated script

### CODE REVIEW

File: chart/compass/charts/external-services-mock/templates/oauth-credentials-secret.yaml
Comment (PetarTodorovv) [inline chart/compass/charts/external-services-mock/templates/oauth-credentials-secret.yaml:24]: maybe you can use the same thing from https://github.com/kyma-incubator/compass/pull/2198 - `{{ printf "https://%s.%s:%s" .Values.global.externalServicesMock.certSecuredHost .Values.global.ingress.domainName (.Values.service.certPort | toString) | b64enc | quote }}`

File: installation/resources/installer-config-gke-benchmark.yaml.tpl
Comment (la4ezar) [reply installation/resources/installer-config-gke-benchmark.yaml.tpl:28]: The certs and keys are the same in all PRs(we double-checked).
We need this cert in every PR because they are needed for the tests and we don't want to wait for the other PR to be merged.

File: chart/compass/charts/external-services-mock/values.yaml
Comment (la4ezar) [reply chart/compass/charts/external-services-mock/values.yaml:12]: Same as above. We don't want to wait for other PR to be merged.

Comment (PetarTodorovv) [reply chart/compass/charts/external-services-mock/values.yaml:12]: I guess then it will be merged with the first merged PR and should be removed from the others

File: installation/resources/installer-config-gke-benchmark.yaml.tpl
Comment (PetarTodorovv) [reply installation/resources/installer-config-gke-benchmark.yaml.tpl:28]: offline discussion about the certificate and decide to merge this PR and after that to check if we need to add "CA capabilities" to the cert due to merge conflicts

**Review by PetarTodorovv — APPROVED** [review]

**Review by PetarTodorovv — APPROVED** [review]

**Pull request merged**
